### PR TITLE
Dls 13355

### PR DESCRIPTION
**Pull request by ishitaanare**
Radio button defaults - packaging sites add-to-list [P1]

### CODE REVIEW

File: test/controllers/PackagingSiteDetailsControllerSpec.scala [hunk 1]
```diff
@@ -128,10 +128,10 @@ class PackagingSiteDetailsControllerSpec extends SpecBase with MockitoSugar with
         status(result) mustEqual OK
         val page = Jsoup.parse(contentAsString(result))
 
-        contentAsString(result) mustEqual view(form.fill(true), NormalMode, packagingSiteList)(using request, messages(application)).toString
+        contentAsString(result) mustEqual view(form, NormalMode, packagingSiteList)(using request, messages(application)).toString
         page.title() must include("You added 1 packaging site")
         page.getElementsByTag("h1").text() mustEqual "You added 1 packaging site"
-        page.getElementById("value").`val`() mustEqual "true"
+
```
Comment (GujjalapudiA) [inline test/controllers/PackagingSiteDetailsControllerSpec.scala:134]: The test currently confirms that the page displays with the empty form, implying that no radio buttons should be selected. However, it doesn't directly verify the HTML radio inputs. A more explicit assertion would involve inspecting the rendered page to ensure that, when the user revisits, neither Yes nor No radio button is preselected. Something like this would improve clarity.

page.select("input[name=value][value=true]").hasAttr("checked") mustBe false
page.select("input[name=value][value=false]").hasAttr("checked") mustBe false

Comment (mahesh-hmrc) [reply test/controllers/PackagingSiteDetailsControllerSpec.scala:134]: This shouldn't be the approach for testing this. We are not testing empty form behaviour which already implies no selection will be done. If we start adding tests for default behaviour of components, our repos will be bloated by duplicate tests of these components which are already tested elsewhere.

cc: @ishitaanare

**Review by mahesh-hmrc — CHANGES_REQUESTED** [review]
Change requested in this story is to clear the radio button answer once it has been submitted, so that when the user returns to Packaging Site Details to add another packaging site, neither radio button is selected.

Current change only targets `GET` behaviour by always rendering an empty form. That hides the previous answer from the UI, but it does not clear the submitted answer from UserAnswers. On `POST`, when the user selects `Yes`, the controller still stores `PackagingSiteDetailsPage = true` and persists it before ramping to ALF. That means the old `Yes` answer remains in session after it has already been acted on. This is an `add-to-list` pattern, which means `Yes` is a transient action to add the next packaging site. So, once the user is sent to ALF, that answer should be removed before they return to the page and answer the add-another question again.

Integration test currently still expects `"packagingSiteDetails" -> true` to be present in the database after selecting `Yes`, which proves the answer is not being cleared. Ideal fix for this is to remove `PackagingSiteDetailsPage` from UserAnswers after a `Yes` submission before redirecting to ALF. However, a `No` submission should still be persisted, because that is the user confirming they are finished adding packaging sites.

**Comment by ishitaanare** [comment]
Thanks for the feedback! @GujjalapudiA

**Comment by ishitaanare** [comment]
Thanks! @mahesh-hmrc! That makes total sense. I'll update the code and push the changes shortly.

**Comment by ishitaanare** [comment]
Another PR has raised
